sub-packages/bionemo-geometric/src/bionemo/geometric/atom_featurizers.py

```diff
@@ -0,0 +1,512 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from pathlib import Path
+from typing import Iterable, Optional
+
+import pandas as pd
+import torch
+from rdkit import Chem
+from rdkit.Chem import Mol, rdMolDescriptors
+from rdkit.Chem.rdchem import ChiralType, HybridizationType
+from rdkit.Chem.Scaffolds import MurckoScaffold
+
+from bionemo.geometric.base_featurizer import (
+    BaseAtomFeaturizer,
+)
+
+
+ALL_ATOM_FEATURIZERS = [
+    "PeriodicTableFeaturizer",
+    "ElectronicPropertyFeaturizer",
+    "ScaffoldFeaturizer",
+    "SmartsFeaturizer",
+    "CrippenFeaturizer",
+]
+
+
+class AtomicNumberFeaturizer(BaseAtomFeaturizer):
+    """Class for featurizing atom by its atomic number."""
+
+    def __init__(self, max_atomic_num: Optional[int] = None) -> None:
+        """Initializes AtomicNumberFeaturizer class."""
+        MAX_ATOMIC_NUM = 118
+        self.max_atomic_num = max_atomic_num if max_atomic_num else MAX_ATOMIC_NUM
+
+    def n_dim(self) -> int:
+        """Returns dimensionality of the computed features."""
+        return self.max_atomic_num
+
+    def get_atom_features(self, mol: Mol, atom_indices: Optional[Iterable] = None) -> list[int]:
```

Comment:
I want to store all attributes related to atomic number encoding in the class itself. The one-hot featurizer would take information from `n_dim` to do one-hot encoding.
Similarly, for other classes which need scaling of features, it would also provide `min_val` and `max_val`. Does that make sense?

Reply:
If it is used for one hot encoding, why not one-hot encoding be part of this class? And same for min_val and max_val for normalization, the related methods should be part of this type of class.